Classes/Utility/AttributeUtility.php

```diff
@@ -41,6 +41,58 @@ public static function findAttribute(int $attributeId, string $selectFields = '*
         );
     }
 
+    /**
+     * Returns all attribute id's for corresponding product.
+     *
+     * @param int $productId
+     * @param string $selectFields
+     * @return array|null
+     */
+    public static function findAllAttributesForProduct(
+        int $productId,
+        array $attributeIds = [],
+        array $attributeIdentifiers = [],
+        string $selectFields = '*'
+    ): ?array {
+        $queryBuilder = self::getQueryBuilderForTable(AttributeValueRepository::TABLE_NAME);
+        $selectFields = array_map(function ($field) use ($selectFields) {
+            return $selectFields !== '*' ? 'a.' . $field : $field;
+        }, GeneralUtility::trimExplode(',', $selectFields, true));
+
+        $queryBuilder
+            ->select(...$selectFields)
+            ->from(AttributeValueRepository::TABLE_NAME)
+            ->join(
+                AttributeValueRepository::TABLE_NAME,
+                AttributeRepository::TABLE_NAME,
+                'a',
+                $queryBuilder->expr()->eq('a.uid', AttributeValueRepository::TABLE_NAME . '.attribute')
+            )
+            ->where(
+                $queryBuilder->expr()->eq(AttributeValueRepository::TABLE_NAME . '.product', $productId)
+            );
+
+        if (!empty($attributeIds)) {
+            $queryBuilder->andWhere(
+                $queryBuilder->expr()->in(AttributeValueRepository::TABLE_NAME . '.attribute', $attributeIds)
+            );
+        }
+
+        if (!empty($attributeIdentifiers)) {
+            $queryBuilder->andWhere(
+                $queryBuilder->expr()->in('a.identifier', $attributeIdentifiers)
+            );
+        }
+
+        $attributes = $queryBuilder->execute()->fetchAllAssociativeIndexed();
+
+        if (is_array($attributes)) {
+            return array_keys($attributes);
```

Comment:
If you have `$selectFields`, this doesn't make sense. I suspect you should do array_column for whatever info you want to retrieve (like UID), but it should be done outside of this method because you should return all the `$selectFields`.